src/utils/supergraph_config.rs

```diff
@@ -0,0 +1,1169 @@
+use std::str::FromStr;
+
+use anyhow::anyhow;
+use apollo_federation_types::build::{BuildError, BuildErrors, SubgraphDefinition};
+use apollo_federation_types::config::{
+    FederationVersion, SchemaSource, SubgraphConfig, SupergraphConfig,
+};
+use apollo_parser::{cst, Parser};
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+
+use rover_client::blocking::{GraphQLClient, StudioClient};
+use rover_client::operations::subgraph;
+use rover_client::operations::subgraph::fetch::SubgraphFetchInput;
+use rover_client::operations::subgraph::fetch_all::SubgraphFetchAllInput;
+use rover_client::operations::subgraph::introspect::SubgraphIntrospectInput;
+use rover_client::operations::subgraph::{fetch, introspect};
+use rover_client::shared::GraphRef;
+use rover_client::RoverClientError;
+use rover_std::{Fs, Style};
+
+use crate::options::ProfileOpt;
+use crate::utils::client::StudioClientConfig;
+use crate::utils::expansion::expand;
+use crate::utils::parsers::FileDescriptorType;
+use crate::{RoverError, RoverErrorSuggestion, RoverResult};
+
+/// Nominal type that captures the behavior of collecting remote subgraphs into a
+/// [`SupergraphConfig`] representation
+#[derive(Clone, Debug)]
+pub struct RemoteSubgraphs(SupergraphConfig);
+
+impl RemoteSubgraphs {
+    /// Fetches [`RemoteSubgraphs`] from Studio
+    pub fn fetch(
+        client: &StudioClient,
+        federation_version: &FederationVersion,
+        graph_ref: &GraphRef,
+    ) -> RoverResult<RemoteSubgraphs> {
+        let subgraphs = subgraph::fetch_all::run(
+            SubgraphFetchAllInput {
+                graph_ref: graph_ref.clone(),
+            },
+            client,
+        )?;
```

Comment:
this looks like where we'd have our side of the pagination; calling out more as a pointer for myself if we end up talking about pagination more